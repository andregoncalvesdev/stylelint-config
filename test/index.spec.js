/* eslint quote-props:0 */

'use strict';

const path = require('path');
const stylelint = require('stylelint');
const glob = require('glob');

const files = glob.sync(`${__dirname}/fixtures/**/*.css`);

files.forEach((file) =>
    it(`should pass on ${path.relative(__dirname, file)}`, async () => {
        const result = await stylelint.lint({ files: file });
        const output = JSON.parse(result.output);

        output.forEach((entry) => {
            const result = entry.warnings
            .sort((warn1, warn2) => warn1.line - warn2.line || warn1.column - warn2.column || warn1.rule.localeCompare(warn2.rule))
            .map((warn) => ({ rule: warn.rule, severity: warn.severity, line: warn.line, column: warn.column }));

            expect(result).toMatchSnapshot();
        });
    })
);
