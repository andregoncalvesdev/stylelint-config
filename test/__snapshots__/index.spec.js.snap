// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass on fixtures/basic.bad.css 1`] = `
Array [
  Object {
    "column": 9,
    "line": 2,
    "rule": "string-quotes",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 3,
    "rule": "at-rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 5,
    "rule": "at-rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 14,
    "rule": "rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 15,
    "rule": "comment-no-empty",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 21,
    "rule": "rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 25,
    "rule": "at-rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 30,
    "rule": "comment-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 33,
    "rule": "rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 36,
    "rule": "comment-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 38,
    "rule": "rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 41,
    "rule": "comment-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 42,
    "rule": "rule-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 45,
    "rule": "comment-empty-line-before",
    "severity": "error",
  },
  Object {
    "column": 1,
    "line": 49,
    "rule": "comment-empty-line-before",
    "severity": "error",
  },
]
`;

exports[`should pass on fixtures/basic.good.css 1`] = `Array []`;

exports[`should pass on fixtures/order.bad.css 1`] = `
Array [
  Object {
    "column": 5,
    "line": 7,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 8,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 14,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 20,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 23,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 27,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 32,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 34,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 37,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 43,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 43,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 56,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 62,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 66,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 70,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 76,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 83,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 83,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 95,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 104,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 112,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 118,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 124,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 124,
    "rule": "order/properties-order",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 134,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
  },
  Object {
    "column": 5,
    "line": 134,
    "rule": "order/properties-order",
    "severity": "error",
  },
]
`;

exports[`should pass on fixtures/order.good.css 1`] = `Array []`;
